lib/index.js

```diff
@@ -50,87 +51,92 @@ function getMinimumLimit(limit1, limit2) {
   return limit2;
 }
 
-function setupRateLimitEventExt(server, options) {
-  const event = options.event;
-  const extractKey = options.extractKey;
-  const onError = options.onError;
-
-  const extractKeyAndTakeToken = function(limitd, request, reply, type) {
-    extractKey(request, reply, (err, key) =>{
-      if (err) { return reply(err); }
-
-      if (!limitd) {
-        // limitd is not connected, do not fail!
-        return reply.continue();
-      }
-
-      limitd.take(type, key, (err, currentLimitResponse) => {
-        if (err){
-          if (onError) { return onError(err, reply); }
-          // by default we don't fail if limitd is unavailable
-          return reply.continue();
-        }
-
-        const oldMinimumLimitResponse = request.plugins.patova && request.plugins.patova.limit
-        const newMinimumLimitResponse = getMinimumLimit(currentLimitResponse, oldMinimumLimitResponse)
+function buildRateLimitEvent({ extractKey, limitd, onError, type }) {
+  async function getType(request) {
+    if (typeof type !== 'function') {
+      return Promise.resolve(type);
```

Comment:
You can just `return type;` here.

Reply:
I wanted to honor the interface :) But make sense to make it simple